lib/matplotlib/ticker.py

```diff
@@ -902,14 +902,14 @@ def __call__(self, x, pos=None):
         fx = math.log(abs(x)) / math.log(b)
         isDecade = is_close_to_int(fx)
         exponent = np.round(fx) if isDecade else np.floor(fx)
-        coeff = np.round(x / b ** exponent)
+        coeff = np.round(abs(x) / b ** exponent)
         if coeff in self.sublabel:
             if not isDecade and self.labelOnlyBase:
-                s = ''
+                return ''
             elif x > 10000:
-                s = '%1.0e' % x
+                s = '%1.0e' % abs(x)
```

Comment:
Something is still wrong--this doesn't make sense.  It looks like the solution is to remove all of these `abs` calls, and instead set `x = abs(x)` right after saving the sign of  x, a few lines above.